external-crates/move/crates/move-package-alt/src/compatibility/legacy_lockfile.rs

```diff
@@ -49,15 +51,35 @@ pub fn load_legacy_lockfile(
         return Ok(None);
     }
 
+    // The old package system didn't fail if the lockfile version field was too high, instead it
+    // just mushed the lockfile fields in with the new lockfile fields. We detect this case and
+    // complain loudly (and preserve the modern pins)
+    if let Some(pinned) = lockfile.get("pinned") {
+        warn!(
+            "{}: Detected a modern lockfile at {lockfile_path:?} that was modified by an older CLI; some information may be lost. Be sure that all contributors are using the latest CLI.",
```

Comment:
```suggestion
            "{}: Detected a modern lockfile at {lockfile_path:?} that was modified by an older CLI; some information may be lost. Be sure that all contributors are using the latest CLI.",
```

hard to force all contributors to use the same CLI, particularly if this would be an open source project.